src/scan-options.ts

```diff
@@ -1,22 +1,51 @@
-export interface ScanOptions {
+/** Options for [[scan]] and [[scanAll]] */
+export type ScanOptions = ScanIndexOptions | ScanNoIndexOptions;
+
+/**
+ * Options for [[scan]] and [[scanAll]] when scanning over the entire key space.
+ */
+export type ScanNoIndexOptions = {
   /** Only include keys starting with `prefix`. */
   prefix?: string;
 
-  /**
-   * The first key to include in the result. If `startKeyExclusive` is `true`
-   * then this key is not included and the result start withthe next larger key.
-   */
-  startKey?: string;
+  /** Only include up to `limit` results. */
+  limit?: number;
+
+  /** Do a `scan` over a named index. */
+  indexName?: undefined;
```

Comment:
Shouldn't this be removed in the NoIndexOptions case?

Reply:
It should, I was having TS issues. Let me try again.